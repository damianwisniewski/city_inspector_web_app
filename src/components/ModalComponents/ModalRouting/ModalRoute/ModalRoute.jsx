import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { ModalContext } from '../ModalRouter/ModalRouter'
import Modal from '../../Modal/Modal'

class ModalRoute extends Component {
	static contextType = ModalContext

	render() {
		const { component, closeButton, darkOverlay } = this.props
		const { handleCloseModal, additionalProps } = this.context

		return (
			<Modal
				closeButton={closeButton}
				darkOverlay={darkOverlay}
				onOverlayClick={handleCloseModal}
				onButtonClose={handleCloseModal}
				{...additionalProps}
			>
				{React.createElement(component, { closeModal: handleCloseModal })}
			</Modal>
		)
	}
}

ModalRoute.defaultProps = {
	closeButton: false,
	darkOverlay: false,
}

ModalRoute.propTypes = {
	component: PropTypes.elementType.isRequired,
	closeButton: PropTypes.bool,
	darkOverlay: PropTypes.bool,
}

export default ModalRoute
