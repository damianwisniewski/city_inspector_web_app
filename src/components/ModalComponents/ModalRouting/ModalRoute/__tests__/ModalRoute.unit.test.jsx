import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import ModalRoute from '../ModalRoute'

configure({ adapter: new Adapter() })

describe('<ModalRoute />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let ModalRouteComponent
	const MockComponent = () => <p>mock</p>

	test('should render correctly', () => {
		ModalRouteComponent = shallow(<ModalRoute component={MockComponent} />)

		expect(toJson(ModalRouteComponent)).toMatchSnapshot()
	})

	test('should provide darkOverlay and closeButton to Modal', () => {
		ModalRouteComponent = shallow(<ModalRoute component={MockComponent} closeButton darkOverlay />)
		expect(ModalRouteComponent.find('Modal').props()).toMatchObject({
			closeButton: true,
			darkOverlay: true,
		})
	})

	//TODO: tests related to context
})
