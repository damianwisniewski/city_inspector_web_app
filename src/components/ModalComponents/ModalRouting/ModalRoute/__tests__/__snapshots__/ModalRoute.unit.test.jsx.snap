// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalRoute /> should render correctly 1`] = `
<Modal
  closeButton={false}
  darkOverlay={false}
>
  <MockComponent />
</Modal>
`;
